ui/src/components/tracks/base_counter_track.ts

```diff
@@ -831,12 +832,45 @@ export abstract class BaseCounterTrack implements TrackRenderer {
         ctx.lineTo(x, lastY);
       }
       lastDrawnY = lastY;
+      lastX = x;
     }
-    ctx.lineTo(endPx, lastDrawnY);
-    ctx.lineTo(endPx, zeroY);
+
+    // Draw the main shape (without extension)
+    // We need to stroke before closing the path to avoid stroking the vertical
+    // line at lastX
+    ctx.lineTo(lastX, lastDrawnY);
+    ctx.stroke();
+
+    // Now close the fill without stroking
+    ctx.lineTo(lastX, zeroY);
     ctx.closePath();
     ctx.fill();
-    ctx.stroke();
+
+    // Draw fade-out from last sample to edge
+    if (endPx > lastX) {
+      // Gradient fill for fade-out (no vertical line at lastX)
+      const gradient = ctx.createLinearGradient(lastX, 0, endPx, 0);
+      gradient.addColorStop(0, `hsla(${hue}, 45%, 50%, 0.6)`);
+      gradient.addColorStop(1, `hsla(${hue}, 45%, 50%, 0)`);
+
+      ctx.fillStyle = gradient;
+      ctx.beginPath();
+      ctx.moveTo(lastX, lastDrawnY);
+      ctx.lineTo(endPx, lastDrawnY);
+      ctx.lineTo(endPx, zeroY);
+      ctx.lineTo(lastX, zeroY);
+      ctx.closePath();
+      ctx.fill();
+
+      // Dashed line
+      ctx.strokeStyle = `hsl(${hue}, 10%, 71%)`;
```

Comment:
Maybe just keep the dashed line the same color as the regular line? It's so desaturated it just looks grey.